### PR TITLE
Added WorkflowPresenter Test

### PR DESCRIPTION
**Pull request by Hiteshgautam01**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the checks with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -166,6 +166,7 @@ limitations under the License.
     <string name="err_login_email">Enter valid email</string>
     <string name="err_login_password">Enter valid password</string>
     <string name="error_failed_to_fetch_workflow">Failed to fetch Workflow</string>
+    <string name="no_workflows_avialable">No workflows avialable</string>
```
Comment (sagar15795) [inline app/src/main/res/values/strings.xml]: make it `More workflows not available`

**Comment by sagar15795** [comment]
@Hiteshgautam01 Please fix Travis and Please don't check things blindly.

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (Hiteshgautam01) [reply app/src/main/res/values/strings.xml]: I am using this when there is no workflow is created by any user on myExperiment. Then this is for empty workflows.

**Comment by Hiteshgautam01** [comment]
Hey @sagar15795 , we can test for multiple pages by using loop and threads but it shows the same result. It is more appropriate to test in UI testing using `ViewActions.scrollTo()` in espresso.

**Comment by Hiteshgautam01** [comment]
Hey @sagar15795 , Travis has been fixed. Please Review.

**Comment by Hiteshgautam01** [comment]
Hey @sagar15795, done the requested changes. Please review.

File: app/src/main/java/org/apache/taverna/mobile/ui/workflow/WorkflowPresenter.java [hunk 2]
```diff
@@ -72,9 +72,17 @@ public void loadAllWorkflow(int pageNumber) {
...
                     @Override
                     public void onNext(Workflows workflows) {
-                        getMvpView().showProgressbar(false);
-                        getMvpView().removeLoadMoreProgressbar();
-                        getMvpView().showWorkflows(workflows);
+                        if (workflows.getWorkflowList() != null) {
+                            getMvpView().showProgressbar(false);
+                            getMvpView().removeLoadMoreProgressbar();
+                            getMvpView().showWorkflows(workflows);
+                        } else if (workflows.getWorkflowList() == null && pageNumber == 1) {
```
Comment (sagar15795) [inline app/src/main/java/org/apache/taverna/mobile/ui/workflow/WorkflowPresenter.java]: make the logic this way.
```
else{
	if(pageNumber == 1){
    	getMvpView().showSnackBar(R.string.no_workflows_found);
	} else {
	 getMvpView().showSnackBar(R.string.no_more_workflows_avialable);
	}
    getMvpView().removeLoadMoreProgressbar();
} 
```

File: app/src/test/java/org/apache/taverna/mobile/ui/workflow/WorkflowPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+    @Before
+    public void setUp() {
+
+        workflowPresenter = new WorkflowPresenter(dataManager);
+        workflowPresenter.attachView(workflowMvpView);
+
+        workflows = FakeRemoteDataSource.getWorkflowList();
+
+        option_pg1 = new HashMap<>();
```
Comment (sagar15795) [inline app/src/test/java/org/apache/taverna/mobile/ui/workflow/WorkflowPresenterTest.java]: use `optionPage1` in place of `option_pg1`

File: app/src/test/java/org/apache/taverna/mobile/ui/workflow/WorkflowPresenterTest.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        option_pg1.put("elements", "title,type,uploader,preview,created-at");
+        option_pg1.put("page", String.valueOf(1));
+        option_pg1.put("num", String.valueOf(10));
+        option_pg1.put("order", "reverse");
+
+        option_pg2 = new HashMap<>();
```
Comment (sagar15795) [inline app/src/test/java/org/apache/taverna/mobile/ui/workflow/WorkflowPresenterTest.java]: use `optionPage2` in place of `option_pg2`

**Pull request merged**